aiidalab_widgets_base/structures.py

```diff
@@ -702,33 +712,63 @@ def _rdkit_opt(self, smiles, steps):
 
         smiles = smiles.replace("[", "").replace("]", "")
         mol = Chem.MolFromSmiles(smiles)
+        if mol is None:
+            self.output.value = (
+                "WARNING: RDKit failed to parse SMILES, trying OpenBabel."
+            )
+            return None
         mol = Chem.AddHs(mol)
 
-        AllChem.EmbedMolecule(mol, maxAttempts=20, randomSeed=42)
+        params = AllChem.ETKDG()
+        params.randomSeed = 42
+        params.maxAttempts = 20
+        conf_id = AllChem.EmbedMolecule(mol, params=params)
+        if conf_id == -1:
+            # useRandomCoords might be more robust for larger molecules
+            # https://sourceforge.net/p/rdkit/mailman/message/21776083/
+            params.useRandomCoords = True
+            conf_id = AllChem.EmbedMolecule(mol, params=params)
+        if conf_id == -1:
+            msg = "WARNING: RDkit embedding failed, trying OpenBabel"
+            self.output.value = msg
+            return None
+
+        if not AllChem.UFFHasAllMoleculeParams(mol):
+            msg = "WARNING: RDKit is missing UFF parameters, trying OpenBabel"
+            self.output.value = msg
+            return None
+
         AllChem.UFFOptimizeMolecule(mol, maxIters=steps)
-        positions = mol.GetConformer().GetPositions()
+        positions = mol.GetConformer(id=conf_id).GetPositions()
         natoms = mol.GetNumAtoms()
         species = [mol.GetAtomWithIdx(j).GetSymbol() for j in range(natoms)]
         return self._make_ase(species, positions, smiles)
 
-    def _mol_from_smiles(self, smiles, steps=10000):
+    def _mol_from_smiles(self, smiles, steps=1000):
         """Convert SMILES to ase structure try rdkit then pybel"""
         try:
-            return self._rdkit_opt(smiles, steps)
-        except ValueError:
-            return self._pybel_opt(smiles, steps)
+            struct = self._rdkit_opt(smiles, steps)
+            if struct is None:
+                struct = self._pybel_opt(smiles, steps)
+            return struct
+        except ValueError as e:
+            self.output.value = str(e)
+            return None
 
     def _on_button_pressed(self, change):  # pylint: disable=unused-argument
         """Convert SMILES to ase structure when button is pressed."""
         self.output.value = ""
 
         if not self.smiles.value:
             return
-        self.output.value = "Screening possible conformers {}".format(
+        spiner = "Screening possible conformers {}".format(
             self.SPINNER
         )  # font-size:20em;
```

Comment:
```suggestion
        spiner =f "Screening possible conformers {self.SPINNER}"  # font-size:20em;
```

Reply:
I do not remember where the .replace("[") was coming from. In any case the goal is to have a ROBUST app to convert smiles into a geometry NOT to have an app tuned for some non-standard case. So GrapheneNanoRibbons are a very particular case that we typically do not handle through SMILES. if the "[" was a trick to help with GNRs it shoudl then be removed.

Reply:
@cpignedoli Could you provide please provide a link if the GNR work was published.